backend/src/test/java/com/google/sps/TestEmailNotifier.java

```diff
@@ -0,0 +1,109 @@
+package com.google.sps;
+
+import com.google.sps.notifs.EmailNotifier;
+import org.junit.Assert;
+import org.junit.Test;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.IOException;
+import java.util.Properties;
+import static org.junit.Assert.*;
+
+public class TestEmailNotifier {
+
```

Comment:
Now that the `Gmail` dependency can be injected through the constructor, this test can provide a mock Gmail instance that can be used to stub / verify interactions with `Gmail`. 

To start, I'd recommend adding a set up method to the test to setup the mock. Once the mock is setup, you can construct an EmailNotifier and make it a field on the test class. (The test class is recreated for each test method so there's no need to worry about state carrying over from one test to the next)

```
private EmailNotifier emailNotifier;

@Before
public void setUp() {
  Gmail gmail = mock(Gmail.class)
   ....

  emailNotifier = new EmailNotifier(...)
}
```

Reply:
Noted